moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planning_manager.cpp

```diff
@@ -184,14 +184,6 @@ bool HybridPlanningManager::sendGlobalPlannerAction()
           ReactionResult reaction_result =
               planner_logic_instance_->react(HybridPlanningEvent::GLOBAL_PLANNING_ACTION_FINISHED);
           auto result = std::make_shared<moveit_msgs::action::HybridPlanner::Result>();
-          if (reaction_result.error_code.val != moveit_msgs::msg::MoveItErrorCodes::SUCCESS)
```

Comment:
This if() was never entered because the value of `reaction_result` is constant (see 2 lines up)

Reply:
> @sjahr I believe that [f8b0cbd](https://github.com/ros-planning/moveit2/commit/f8b0cbd1bb541a2f11f8eb6f0cfeb17d8d19fe24) fixed my comment above about `reaction_result` always reflecting SUCCESS. Take a look if you have time.

The reaction result is not constant but depends on the planner logic used (because it defines the react() function):
```cpp
ReactionResult SinglePlanExecution::react(const HybridPlanningEvent& event)
{
  switch (event)
  {
    case HybridPlanningEvent::HYBRID_PLANNING_REQUEST_RECEIVED:
      if (!hybrid_planning_manager_->sendGlobalPlannerAction())  // Start global planning
      {
        hybrid_planning_manager_->sendHybridPlanningResponse(false);
      }
      break;
    case HybridPlanningEvent::GLOBAL_SOLUTION_AVAILABLE:
      std::call_once(LOCAL_PLANNER_STARTED, [this]() {            // ensure the local planner is not started twice
        if (!hybrid_planning_manager_->sendLocalPlannerAction())  // Start local planning
        {
          hybrid_planning_manager_->sendHybridPlanningResponse(false);
        }
      });
      break;
    case HybridPlanningEvent::LOCAL_PLANNING_ACTION_FINISHED:
      hybrid_planning_manager_->sendHybridPlanningResponse(true);
      break;
    default:
      break;
      // Do nothing
  }
  return ReactionResult(event, "", moveit_msgs::msg::MoveItErrorCodes::SUCCESS); // This never returns failure
}
```

The available planner logic plugins ([single_plan_execution](https://github.com/ros-planning/moveit2/pull/813/files#diff-0b0779bd232dff42badd15757460d5e7d4967d4eb6f42ac09d8076e80e970b85) or replan_invalidated_trajectory) do not return failure at any point but other logic plugins developed in the future might do that.

I think this fix is not correct because the logic should define how the architecture reacts to a failing global/local planner. Instead of aborting it might be desired to rerun the planner multiple times or adapt its constraints. Now it is not possible for the user-defined logic to handle this case because we abort hybrid planning before the react() function is called in case the planner fails. 
The previous implementation in the hybrid planning manager was correct but I guess the plugin logic code above needs a fix. I think the planner result needs to be saved in a variable of the hybrid planning manager and the evaluated in the plannger logic.

Reply:
I'm not sure the `HybridPlanningEvent` enum contains enough information to ever communicate a failure. These are the only options:

```
  // Occurs when the hybrid planning manager receives a planning request
  HYBRID_PLANNING_REQUEST_RECEIVED,
  // Indicates that the global planning action finished (Does not indicate planning success! The action response
  // contains this information)
  GLOBAL_PLANNING_ACTION_FINISHED,
  // Indicates that the global planner found a solution (This solution is not necessarily the last or best solution)
  GLOBAL_SOLUTION_AVAILABLE,
  // Indicates that the local planning action finished (Does not indicate planning success! The action response contains this information)
  LOCAL_PLANNING_ACTION_FINISHED,
```

What would I put in case of a failure?

Reply:
I think it's best if I wait for you to make a PR about this.

Reply:
I think my confusion arose because I had put the "local planner stuck detection" functionality inside LocalPlannerComponent. That's not where it belongs -- it belongs in the local solver plugin, which is ForwardTrajectory. Having made that change, I don't have any issue communicating the failure.